File: Leetcode/ThreeConsecutiveOdds.rs
// https://leetcode.com/problems/three-consecutive-odds

struct Solution {}

impl Solution {
    pub fn three_consecutive_odds(arr: Vec<i32>) -> bool {
        arr.windows(3).any(|v| v[0] & v[1] & v[2] & 1 > 0)
    }
}

File: Leetcode/ThreeConsecutiveOdds.c
// https://leetcode.com/problems/three-consecutive-odds

int threeConsecutiveOdds(int* arr, int arrSize) {
    for (int i = 0; i < arrSize - 2; ++i) {
        if (arr[i] & arr[i + 1] & arr[i + 2] & 1) {
            return 1;
        }
    }
    return 0;
}